ui/org.eclipse.pde.ui.tests/src/org/eclipse/pde/ui/tests/target/IUBundleContainerTests.java

```diff
@@ -327,9 +377,58 @@ public void testExternalModelManagerPreferences() throws Exception {
 	}
 
 	/**
-	 * Creates an IU bundle container with the specified IUs from the test repository.
+	 * Creates an in-memory artifact repository over {@code test/sites/site.a.b}
 	 *
-	 * @param unitIds identifiers of IU's to add to the container
+	 * @return in-memory artifact repository
+	 */
+	protected VirtualArtifactRepository createVirtualArtifactRepository() throws Exception {
+		File repoFolder = new File(getURI("/tests/sites/site.a.b"));
+		// DirectoryBundleContainer expects features to be unpacked
+		File features = new File(repoFolder, "features");
+		for (File jarFile : features.listFiles()) {
+			if (jarFile.isDirectory()) {
+				continue;
+			}
+			String jarFileName = jarFile.getName();
+			String jarFileBaseName = jarFileName.substring(0, jarFileName.length() - ".jar".length());
+
+			File outDir = new File(features, jarFileBaseName);
+			outDir.mkdir();
+			outDir.deleteOnExit();
```

Comment:
I don't think this will be able to delete entire directories with their content.
If you need a temparary directory you can use JUnit's TemporaryFolder, which is reliably cleared depending on if it is a `@Rule` or `@ClassRule`.

Reply:
It's sadly not that simple :/
Even with the temporary folder, I then still need to copy the update site to the test folder. In the end, I simply test against the update site in `modified-jdt-features.zip`, where the features are already unpacked.